src/sentry/api/endpoints/external_user_details.py

```diff
@@ -1,50 +1,47 @@
 import logging
+from typing import Any, Tuple
 
-from django.http import Http404
 from rest_framework import status
-from rest_framework.exceptions import PermissionDenied
 from rest_framework.response import Response
 
+from sentry.api.bases.external_actor import ExternalActorEndpointMixin, ExternalUserSerializer
 from sentry.api.bases.organization import OrganizationEndpoint
 from sentry.api.serializers import serialize
-from sentry.models import ExternalUser
-
-from .external_user import ExternalUserMixin, ExternalUserSerializer
+from sentry.models import ExternalActor, Organization
 
 logger = logging.getLogger(__name__)
 
 
-class ExternalUserDetailsEndpoint(OrganizationEndpoint, ExternalUserMixin):
-    def convert_args(self, request, organization_slug, external_user_id, *args, **kwargs):
-        args, kwargs = super().convert_args(request, organization_slug, *args, **kwargs)
-        try:
-            kwargs["external_user"] = ExternalUser.objects.get(
-                id=external_user_id,
-            )
-        except ExternalUser.DoesNotExist:
-            raise Http404
+class ExternalUserDetailsEndpoint(OrganizationEndpoint, ExternalActorEndpointMixin):
+    def convert_args(self, request: Any, *args: Any, **kwargs: Any) -> Tuple[Any, Any]:
+        """ A little magic to make types work for convert_args. """
+        return self._convert_args(request, *args, **kwargs)
 
-        return (args, kwargs)
+    def _convert_args(
```

Comment:
@mgaeta can we merge this function with `convert_args`? I don't see this function used elsewhere.